File: Exploratory analysis/TP1/Python/TD1.py
import pandas as pd
import numpy as np

from scipy.stats import chi2_contingency

# Importing the csv and loading it in a dataframe.
CSV_FILE_PATH = "../menu.csv"
menu_df = pd.read_csv(CSV_FILE_PATH)


def chisq_of_df_cols(df, c1, c2):
    groupsizes = df.groupby([c1, c2]).size()
    ctsum = groupsizes.unstack(c1)
    # fillna(0) is necessary to remove any NAs which will cause exceptions
    return(chi2_contingency(ctsum.fillna(0)))


FIELDS_OF_INTEREST = ["Calories", "Total Fat",
                      "Cholesterol", "Sodium", "Sugars", "Protein"]

chi2_results_df = pd.DataFrame(len(FIELDS_OF_INTEREST)*[len(FIELDS_OF_INTEREST)*[0]], columns=FIELDS_OF_INTEREST)

for i in range(len(FIELDS_OF_INTEREST) - 1):
    for j in range(i + 1, len(FIELDS_OF_INTEREST)):
        chi2 = chisq_of_df_cols(
            menu_df, FIELDS_OF_INTEREST[i], FIELDS_OF_INTEREST[j])[0]

        chi2_results_df.iloc[i, j] = chi2
        chi2_results_df.iloc[j, i] = chi2

chi2_results_df["index"] = FIELDS_OF_INTEREST
chi2_results_df=chi2_results_df.set_index("index")

print(chi2_results_df)

File: Exploratory analysis/TP1/R/TD1.r
# Load the menu csv as a dataframe.
csv_file_path <- "Exploratory analysis/TP1/menu.csv"
menu_df <- read.csv(csv_file_path)

# Prints the dataframe.
print(menu_df)

# Compute the Chi square test for Calories and Total Fat.
print(chisq.test(menu_df$Calories,
        menu_df$Total.Fat))

# Name of the variables we are interested in computing the Chi square test.
fields_of_interest <- c("Calories", "Total.Fat",
                      "Cholesterol", "Sodium", "Sugars", "Protein")

n <- length(fields_of_interest)

# Initializing the matrix containing the chi square values.
chi2_matrix <- matrix(0, ncol = n, nrow = n, dimnames = list(fields_of_interest, fields_of_interest))

# Computing the Chi square values, and using the symmetry of the matrix.
for (i in 1:(length(fields_of_interest) - 1)) {
    for (j in (i + 1):length(fields_of_interest)) {
        chi2 <- chisq.test(menu_df[, fields_of_interest[i]],
                           menu_df[, fields_of_interest[j]])$statistic

        chi2_matrix[i, j] <- chi2
        chi2_matrix[j, i] <- chi2
    }
}

print(chi2_matrix)

corr_matrix <- round(cor(menu_df[, fields_of_interest]), 2)
print(corr_matrix > 0.5)
print(corr_matrix)

corr_matrix <- round(cor(scale(menu_df[, fields_of_interest])), 2)
print(corr_matrix > 0.5)
print(corr_matrix)

# We can see that all but 'Sugars' are correlated (with r>0,5)
